scenes/game_elements/props/collectible_item/components/collectible_item.gd

```diff
@@ -28,29 +28,19 @@ class_name CollectibleItem extends Node2D
 		collected_dialogue = new_value
 		notify_property_list_changed()
 ## The dialogue title from where [member collected_dialogue] will start.
-var dialogue_title: StringName = ""
+@export var dialogue_title: StringName = ""
 
 @onready var interact_area: InteractArea = $InteractArea
 @onready var animation_player: AnimationPlayer = $AnimationPlayer
 @onready var sprite_2d: Sprite2D = $Sprite2D
 @onready var appear_sound: AudioStreamPlayer = %AppearSound
 
 
-func _get_property_list() -> Array[Dictionary]:
-	var properties: Array[Dictionary] = []
-
-	if collected_dialogue:
-		properties.push_back(
-			{
-				"name": "dialogue_title",
-				"usage": PROPERTY_USAGE_DEFAULT,
-				"type": TYPE_STRING,
-				"hing": PROPERTY_HINT_PLACEHOLDER_TEXT,
-				"hint_string": ""
-			}
-		)
-
-	return properties
+func _validate_property(property: Dictionary) -> void:
+	match property.name:
+		"dialogue_title":
+			if not collected_dialogue:
+				property.usage |= PROPERTY_USAGE_READ_ONLY
```

Comment:
Does the `property` get reset each time this function is called? Or do you need to clear the flag when `collected_dialogue` is `true`?

Reply:
I think that the vaildation happens each time any property changes:
[Grabación de pantalla desde 2025-05-13 14-43-20.webm](https://github.com/user-attachments/assets/fe33e000-4cda-45f2-8651-91df7fbe778c)

Reply:
Got it!